rim/static/rim/js/dynamic_fields.js

```diff
@@ -0,0 +1,99 @@
+        </div>`;
+let nic_form = (id, mac, name, type) => `<div class='small-4 large-4 addcell cell'>
+        <label>Network Card #${id + 1}</label>
+        <input id="nic_mac_${id}" type="text" placeholder="MAC" value="${mac}"/>
+        <input id="nic_name_${id}" type="text" placeholder="Name" value="${name}" />
+        <input id="nic_type_${id}" type="text" placeholder="Type" value="${type}" />
+        </div>`;
+let display_form = (id, height, manufacturer, code, resolution, serial, name, width) => `<div class='small-4 large-4 addcell cell'>
+        <label>Display #${id + 1}</label>
+        <input id="display_height_${id}" type="text" placeholder="Height in MM" value="${height}"/>
+        <input id="display_manufacturer_${id}" type="text" placeholder="Manufacturer" value="${manufacturer}" />
+        <input id="display_code_${id}" type="text" placeholder="Product Code" value="${code}" />
+        <input id="display_resolution_${id}" type="text" placeholder="Resolution" value="${resolution}" />
+        <input id="display_serial_${id}" type="text" placeholder="Serial No" value="${serial}" />
+        <input id="display_name_${id}" type="text" placeholder="Name" value="${name}" />
+        <input id="display_width_${id}" type="text" placeholder="Width in MM" value="${width}" />
+        </div>`;
+
+let storages = 0, gpus = 0, nics = 0, displays = 0;
+
+function populateInputs() {
+    for (const s of json_fields['storage'])
+        document.getElementById('storage_inputs').innerHTML += storage_form(storages++, s['Model'], s['Size']);
+
+    for (const g of json_fields['GPU'])
+        document.getElementById('gpu_inputs').innerHTML += gpu_form(gpus++, g['Name']);
+
+    for (const n of json_fields['network_cards'])
+        document.getElementById('nic_inputs').innerHTML += nic_form(nics++, n['MAC'], n['Name'], n['Type']);
+
+    for (const d of json_fields['displays'])
+        document.getElementById('display_inputs').innerHTML += display_form(displays++, d["HeightInMillimeters"],
+            d["ManufacturerName"], d["ProductCodeID"], d["Resolution"], d["SerialNumberID"], d["UserFriendlyName"],
+            d["WidthInMillimeters"]);
+}
+
+function addInputs(field) {
+    switch (field) {
+        case 'storage': document.getElementById('storage_inputs').innerHTML += storage_form(storages++, '', '');
+            break;
+        case 'gpu': document.getElementById('gpu_inputs').innerHTML += gpu_form(gpus++, '', '');
+            break;
+        case 'nic': document.getElementById('nic_inputs').innerHTML += nic_form(nics++, '', '', '');
+            break;
+        case 'display': document.getElementById('display_inputs').innerHTML += display_form(displays++, '', '', '', '', '', '', '');
+            break;
+
+    }
+}
+
+function submitAddEditForm() {
+    let storage_json = [], gpu_json = [], nic_json = [], display_json = [];
+
+    for (let i = 0; i < storages; i++)
+        if (document.getElementById('storage_model_' + i).value)
```

Comment:
The way this if statement is written, if the user doesn't enter a value for the model then size will not be submitted either. This will delete the size entry in the database if it is the only field present upon saving, even if the user only changes the hostname for example.